lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -1054,29 +1054,29 @@ namespace Js
             scriptContext->ResetWeakReferenceDictionaryList();
             scriptContext->SetIsFinalized();
             scriptContext->GetThreadContext()->UnregisterScriptContext(scriptContext);
+            scriptContext->MarkForClose();
```

Comment:
Is this here just for ScriptContext::SaveStartupProfileAndRelease?  Can you just move that function to JAvascriptLibrary and call it directly?
Then you don't need to add the ISClose check in MarkForClose too, right?